pkg/build/api/validation/validation_test.go

```diff
@@ -213,54 +218,227 @@ func TestBuildConfigValidationFailureRequiredName(t *testing.T) {
 				{
 					Type:        buildapi.ImageChangeBuildTriggerType,
 					ImageChange: &buildapi.ImageChangeTrigger{},
 				},
 				{
+					Type: buildapi.ImageChangeBuildTriggerType,
+					ImageChange: &buildapi.ImageChangeTrigger{
+						From: &kapi.ObjectReference{
+							Kind: "DockerImage",
+							Name: "myimage:tag",
+						},
+					},
+				},
+			},
+			expectError: true,
+			errorType:   fielderrors.ValidationErrorTypeInvalid,
+		},
+		{
+			name: "empty reference kind for trigger",
+			triggers: []buildapi.BuildTriggerPolicy{
+				{
 					Type:        buildapi.ImageChangeBuildTriggerType,
 					ImageChange: &buildapi.ImageChangeTrigger{},
 				},
+				{
+					Type: buildapi.ImageChangeBuildTriggerType,
+					ImageChange: &buildapi.ImageChangeTrigger{
+						From: &kapi.ObjectReference{
+							Name: "myimage:tag",
+						},
+					},
+				},
 			},
+			expectError: false,
+		},
+		{
+			name: "duplicate imagestreamtag references",
+			triggers: []buildapi.BuildTriggerPolicy{
+				{
+					Type: buildapi.ImageChangeBuildTriggerType,
+					ImageChange: &buildapi.ImageChangeTrigger{
+						From: &kapi.ObjectReference{
+							Kind: "ImageStreamTag",
+							Name: "myimage:tag",
+						},
+					},
+				},
+				{
+					Type: buildapi.ImageChangeBuildTriggerType,
+					ImageChange: &buildapi.ImageChangeTrigger{
+						From: &kapi.ObjectReference{
+							Kind: "ImageStreamTag",
+							Name: "myimage:tag",
+						},
+					},
+				},
+			},
+			expectError: true,
+			errorType:   fielderrors.ValidationErrorTypeInvalid,
+		},
+		{
+			name: "imagestreamtag references with same name, different ns",
+			triggers: []buildapi.BuildTriggerPolicy{
+				{
+					Type: buildapi.ImageChangeBuildTriggerType,
+					ImageChange: &buildapi.ImageChangeTrigger{
+						From: &kapi.ObjectReference{
+							Kind:      "ImageStreamTag",
+							Name:      "myimage:tag",
+							Namespace: "ns1",
+						},
+					},
+				},
+				{
+					Type: buildapi.ImageChangeBuildTriggerType,
+					ImageChange: &buildapi.ImageChangeTrigger{
+						From: &kapi.ObjectReference{
+							Kind:      "ImageStreamTag",
+							Name:      "myimage:tag",
+							Namespace: "ns2",
+						},
+					},
+				},
+			},
+			expectError: false,
+		},
+		{
+			name: "imagestreamtag references with same name, same ns",
+			triggers: []buildapi.BuildTriggerPolicy{
+				{
+					Type: buildapi.ImageChangeBuildTriggerType,
+					ImageChange: &buildapi.ImageChangeTrigger{
+						From: &kapi.ObjectReference{
+							Kind:      "ImageStreamTag",
+							Name:      "myimage:tag",
+							Namespace: "ns",
+						},
+					},
+				},
+				{
+					Type: buildapi.ImageChangeBuildTriggerType,
+					ImageChange: &buildapi.ImageChangeTrigger{
+						From: &kapi.ObjectReference{
+							Kind:      "ImageStreamTag",
+							Name:      "myimage:tag",
+							Namespace: "ns",
+						},
+					},
+				},
+			},
+			expectError: true,
+			errorType:   fielderrors.ValidationErrorTypeInvalid,
```

Comment:
should have a test case for the scenario i described above...   2 triggers, one that implicitly references the builder image (default trigger) and one that explicitly references it.  error because they ultimately are pointing to the same place.